### DISCUSSION TITLE
When I build it gives me a whole bunch of errors

### DISCUSSION THREAD

**Discussion by Rubyboat1207** [KadeArchive/Kade-Engine#338] [Q&A] [answered]
title
`source/ModchartState.hx:237: characters 9-12 : Type not found : Lua
source/ModchartState.hx:238: characters 3-6 : Unknown identifier : lua
source/ModchartState.hx:246: characters 5-8 : Unknown identifier : lua
source/ModchartState.hx:247: characters 5-9 : Type not found : LuaL
source/ModchartState.hx:248: characters 29-32 : Type not found : Lua
source/ModchartState.hx:249: characters 32-35 : Type not found : Lua
source/ModchartState.hx:250: characters 5-8 : Type not found : Lua
source/ModchartState.hx:252: characters 18-22 : Type not found : LuaL
source/ModchartState.hx:256: characters 64-67 : Type not found : Lua
source/ModchartState.hx:257: characters 6-9 : Unknown identifier : lua
source/ModchartState.hx:263: characters 5-11 : Unknown identifier : setVar
source/ModchartState.hx:264: characters 5-11 : Unknown identifier : setVar
source/ModchartState.hx:265: characters 5-11 : Unknown identifier : setVar
source/ModchartState.hx:266: characters 5-11 : Unknown identifier : setVar
source/ModchartState.hx:267: characters 5-11 : Unknown identifier : setVar
source/ModchartState.hx:269: characters 5-11 : Unknown identifier : setVar
source/ModchartState.hx:270: characters 5-11 : Unknown identifier : setVar
source/ModchartState.hx:271: characters 5-11 : Unknown identifier : setVar
source/ModchartState.hx:272: characters 5-11 : Unknown identifier : setVar
source/ModchartState.hx:274: characters 5-11 : Unknown identifier : setVar
source/ModchartState.hx:275: characters 5-11 : Unknown identifier : setVar
source/ModchartState.hx:277: characters 5-11 : Unknown identifier : setVar
source/ModchartState.hx:278: characters 5-11 : Unknown identifier : setVar
source/ModchartState.hx:280: characters 5-11 : Unknown identifier : setVar
source/ModchartState.hx:281: characters 5-11 : Unknown identifier : setVar
source/ModchartState.hx:283: characters 5-11 : Unknown identifier : setVar
source/ModchartState.hx:284: characters 5-11 : Unknown identifier : setVar
source/ModchartState.hx:285: characters 5-11 : Unknown identifier : setVar
source/ModchartState.hx:287: characters 5-11 : Unknown identifier : setVar
source/ModchartState.hx:288: characters 5-11 : Unknown identifier : setVar
source/ModchartState.hx:289: characters 5-11 : Unknown identifier : setVar
source/ModchartState.hx:290: characters 5-11 : Unknown identifier : setVar
source/ModchartState.hx:292: characters 5-11 : Unknown identifier : setVar
source/ModchartState.hx:294: characters 5-11 : Unknown identifier : setVar
source/ModchartState.hx:300: characters 11-21 : Type not found : Lua_helper
source/ModchartState.hx:302: characters 5-15 : Type not found : Lua_helper
source/ModchartState.hx:312: characters 11-21 : Type not found : Lua_helper
source/ModchartState.hx:316: characters 11-21 : Type not found : Lua_helper
source/ModchartState.hx:321: characters 11-21 : Type not found : Lua_helper
source/ModchartState.hx:325: characters 11-21 : Type not found : Lua_helper
source/ModchartState.hx:329: characters 11-21 : Type not found : Lua_helper
source/ModchartState.hx:334: characters 11-21 : Type not found : Lua_helper
source/ModchartState.hx:338: characters 11-21 : Type not found : Lua_helper
source/ModchartState.hx:342: characters 11-21 : Type not found : Lua_helper
source/ModchartState.hx:346: characters 11-21 : Type not found : Lua_helper
source/ModchartState.hx:352: characters 11-21 : Type not found : Lua_helper
source/ModchartState.hx:356: characters 11-21 : Type not found : Lua_helper
source/ModchartState.hx:360: characters 11-21 : Type not found : Lua_helper
source/ModchartState.hx:364: characters 11-21 : Type not found : Lua_helper
source/ModchartState.hx:368: characters 11-21 : Type not found : Lua_helper
source/ModchartState.hx:372: characters 11-21 : Type not found : Lua_helper
source/ModchartState.hx:377: characters 11-21 : Type not found : Lua_helper
source/ModchartState.hx:381: characters 11-21 : Type not found : Lua_helper
source/ModchartState.hx:385: characters 11-21 : Type not found : Lua_helper
source/ModchartState.hx:389: characters 11-21 : Type not found : Lua_helper
source/ModchartState.hx:395: characters 11-21 : Type not found : Lua_helper
source/ModchartState.hx:399: characters 11-21 : Type not found : Lua_helper
source/ModchartState.hx:403: characters 11-21 : Type not found : Lua_helper
source/ModchartState.hx:407: characters 11-21 : Type not found : Lua_helper
source/ModchartState.hx:418: characters 11-21 : Type not found : Lua_helper
source/ModchartState.hx:423: characters 11-21 : Type not found : Lua_helper
source/ModchartState.hx:428: characters 11-21 : Type not found : Lua_helper
source/ModchartState.hx:433: characters 11-21 : Type not found : Lua_helper
source/ModchartState.hx:438: characters 11-21 : Type not found : Lua_helper
source/ModchartState.hx:443: characters 11-21 : Type not found : Lua_helper
source/ModchartState.hx:447: characters 11-21 : Type not found : Lua_helper
source/ModchartState.hx:451: characters 11-21 : Type not found : Lua_helper
source/ModchartState.hx:455: characters 11-21 : Type not found : Lua_helper
source/ModchartState.hx:459: characters 11-21 : Type not found : Lua_helper
source/ModchartState.hx:464: characters 11-21 : Type not found : Lua_helper
source/ModchartState.hx:468: characters 11-21 : Type not found : Lua_helper
source/ModchartState.hx:472: characters 11-21 : Type not found : Lua_helper
source/ModchartState.hx:476: characters 11-21 : Type not found : Lua_helper
source/ModchartState.hx:480: characters 11-21 : Type not found : Lua_helper
source/ModchartState.hx:484: characters 11-21 : Type not found : Lua_helper
source/ModchartState.hx:491: characters 5-15 : Type not found : Lua_helper
source/ModchartState.hx:495: characters 5-15 : Type not found : Lua_helper
source/ModchartState.hx:499: characters 5-15 : Type not found : Lua_helper
source/ModchartState.hx:503: characters 5-15 : Type not found : Lua_helper
source/ModchartState.hx:507: characters 5-15 : Type not found : Lua_helper
source/ModchartState.hx:511: characters 5-15 : Type not found : Lua_helper
source/ModchartState.hx:515: characters 5-15 : Type not found : Lua_helper
source/ModchartState.hx:519: characters 5-15 : Type not found : Lua_helper
source/ModchartState.hx:523: characters 5-15 : Type not found : Lua_helper
source/ModchartState.hx:527: characters 5-15 : Type not found : Lua_helper
source/ModchartState.hx:531: characters 5-15 : Type not found : Lua_helper
source/ModchartState.hx:535: characters 5-15 : Type not found : Lua_helper
source/ModchartState.hx:590: characters 6-12 : Unknown identifier : setVar
source/ModchartState.hx:592: characters 6-12 : Unknown identifier : setVar
source/ModchartState.hx:594: characters 6-12 : Unknown identifier : setVar
source/ModchartState.hx:601: characters 16-19 : Type not found : Lua`

Its just complaining about lua and I dont know why.
if anyone could help (or just give an html5 build) that would be nice thanks!

**Comment by isakube** [comment]
1. What version?
2. You're trying to compile to HTML?
3. This might be happening because of a certain commit that put modchart functions behind an "#if windows". So probably just compile the current source to fix it.

**Reply by Rubyboat1207** [reply]
1. Latest
2. I want to host it on my website
3. when you say "current source" you mean the download code button or? I did look at the code and it does have a lot of "#if windows" functions I tried just removing them, to no avail.

To specify more, it compiles just fine just that when opening the index page on my server its just a black screen no haxe logo or anything. (I do know how to upload it because I used to have normal fnf on my server and it worked fine)

**Comment by puyoxyz** [comment, marked as answer]
There are already automatic HTML5 builds at https://funkin.puyo.xyz if you just want to play.

If you still want to compile it yourself, download the source code again (or `git pull` if you used git), this has already been fixed.
